lib/instrumentation/modules/cassandra-driver.js

```diff
@@ -0,0 +1,186 @@
+'use strict'
+
+const semver = require('semver')
+const sqlSummary = require('sql-summary')
+
+const shimmer = require('../shimmer')
+
+module.exports = function (cassandra, agent, version, enabled) {
+  if (!enabled) return cassandra
+  if (!semver.satisfies(version, '^3.0.0')) {
+    agent.logger.debug('cassandra-driver version %s not supported - aborting...', version)
+    return cassandra
+  }
+
+  if (cassandra.Client) {
+    shimmer.wrap(cassandra.Client.prototype, 'connect', wrapConnect)
+    shimmer.wrap(cassandra.Client.prototype, 'execute', wrapExecute)
+    shimmer.wrap(cassandra.Client.prototype, 'eachRow', wrapEachRow)
+    shimmer.wrap(cassandra.Client.prototype, 'batch', wrapBatch)
+  }
+
+  return cassandra
+
+  function wrapConnect (original) {
+    return function wrappedConnect (callback) {
+      const span = agent.buildSpan()
+      if (!span) {
+        return original.apply(this, arguments)
+      }
+
+      span.type = 'db.cassandra.connect'
+      span.start()
```

Comment:
This span isn't named (it will fall back to the default name `unnamed`). We should coordinate with the Python agent and agree on a name.

Btw, you can just give the name and type as arguments to `start()` instead of setting them as properties, eg:

```js
span.start('Cassandra: Connect', 'db.cassandra.connect')
```

cc @beniwohli

Reply:
I wasn't sure what to call it, since there's no query to summarize for the name. I'm good with whatever. 🤷‍♂️

Reply:
in this case, we fall back to the class/method name that we wrap, e.g. `Cluster.connect`

Reply:
The Python agent prefixes spans with the name of the class and function that was used to start the span. So in the case of here the span would be called `Cluster.connect` in Python. I think we for now just should call it `Cassandra: Connect`. That aligns with how we name the Elasticsearch queries. Aligning across agents will be a bigger task and should probably not hold back this PR.

Reply:
fixed